modules/hal_gpio/src/gpio.cpp

```diff
@@ -30,67 +30,68 @@
+        }
+    }
 
-void SetGPIOPinFunction(GPIOPort port_num, uint8_t pin_num, GPIOPinFunction pin_function) {
-  PORT->Group[port_num].PINCFG[pin_num].bit.PMUXEN = 0x01;  // Enable the pin mux function
-  // There is a seperate pin mux for even and odd pin numbers (See datasheet)
-  if (pin_num % 2 == 0) {
-    PORT->Group[port_num].PMUX[pin_num >> 1].bit.PMUXE |= pin_function;
-  } else {
-    PORT->Group[port_num].PMUX[pin_num >> 1].bit.PMUXO |= pin_function;
-  }
-}
+    void SetGPIOPinFunction(GPIOPort port_num, uint8_t pin_num, GPIOPinFunction pin_function) {
+        PORT->Group[port_num].PINCFG[pin_num].bit.PMUXEN = 0x01;  // Enable the pin mux function
+        // There is a seperate pin mux for even and odd pin numbers (See datasheet)
+        if (pin_num % 2 == 0) {
+            PORT->Group[port_num].PMUX[pin_num].bit.PMUXE |= pin_function;
+        } else {
+            PORT->Group[port_num].PMUX[pin_num].bit.PMUXO |= pin_function;
+        }
+    }
 
-void SetGPIOPinSamplingMode(GPIOPort port_num, uint8_t pin_num, GPIOSamplingMode sampling_mode) {
-  if (sampling_mode == kGPIOSampleContinuously) {
-    PORT->Group[port_num].CTRL.reg |= 1 << pin_num;
-  } else {
-    PORT->Group[port_num].CTRL.reg &= ~(1 << pin_num);
-  }
-}
+    void SetGPIOPinSamplingMode(GPIOPort port_num, uint8_t pin_num, GPIOSamplingMode sampling_mode) {
+        if (sampling_mode == kGPIOSampleContinuously) {
+            PORT->Group[port_num].CTRL.reg |= 1 << pin_num;
+        } else {
+            PORT->Group[port_num].CTRL.reg &= ~(1 << pin_num);
+        }
+    }
 
-uint8_t GetGPIOPinLevel(GPIOPort port_num, uint8_t pin_num) {
-  return (PORT->Group[port_num].OUT.reg >> pin_num) & 1;
-}
+    uint8_t GetGPIOPinLevel(GPIOPort port_num, uint8_t pin_num) {
+        return (PORT->Group[port_num].OUT.reg >> pin_num) & 1;
+    }
 
-uint8_t GetGPIOPinDirection(GPIOPort port_num, uint8_t pin_num) {
-  return (PORT->Group[port_num].DIR.reg >> pin_num) & 1;
-}
+    uint8_t GetGPIOPinDirection(GPIOPort port_num, uint8_t pin_num) {
+        return (PORT->Group[port_num].DIR.reg >> pin_num) & 1;
+    }
 
-uint8_t GetGPIODriveStrength(GPIOPort port_num, uint8_t pin_num) {
-  return (PORT->Group[port_num].PINCFG[pin_num].bit.DRVSTR);
-}
+    uint8_t GetGPIODriveStrength(GPIOPort port_num, uint8_t pin_num) {
+        return (PORT->Group[port_num].PINCFG[pin_num].bit.DRVSTR);
+    }
+
+    uint8_t GetGPIOPinFunction(GPIOPort port_num, uint8_t pin_num) {
+        if (pin_num % 2 == 0) {
+            return PORT->Group[port_num].PMUX[pin_num].bit.PMUXE;
+        } else {
+            return PORT->Group[port_num].PMUX[pin_num].bit.PMUXO;
+        }
+    }
 
-uint8_t GetGPIOPinFunction(GPIOPort port_num, uint8_t pin_num) {
-  if (pin_num % 2 == 0) {
-    return PORT->Group[port_num].PMUX[pin_num >> 1].bit.PMUXE;
-  } else {
-    return PORT->Group[port_num].PMUX[pin_num >> 1].bit.PMUXO;
-  }
-}
 }  // namespace hal::gpio
```

Comment:
According the google style guide an empty line should be added to end of the file